src/Uno.UI/UI/Xaml/BorderHelper.skia.cs

```diff
@@ -45,5 +46,33 @@ public static void UpdateAllBorderProperties(this IBorderInfoProvider @this)
 		@this.UpdateBackgroundSizing();
 		@this.UpdateBorderThickness();
 	}
+
+	public static void SetUpBrushTransitionIfAllowed(BorderVisual visual, Brush fromBrush, Brush toBrush, BrushTransition transition)
+	{
+		if (transition is null)
+		{
+			return;
+		}
+
+		var oldBrush = fromBrush as SolidColorBrush;
+		var newBrush = toBrush as SolidColorBrush;
+
+		var oldBrushIsNullOrSolidColorBrush = oldBrush is not null || fromBrush is null;
```

Comment:
the variable is called `oldBrushIsNull` but it checks for `oldBrush is not null`

Reply:
This part is ported from WinUI actually